rfc/0014-attribute-based-valuetypes/README.md

```diff
@@ -42,47 +37,61 @@ valuetype CorrelationCoefficient oftype decimal {
 }
 ```
 
-This syntax is smart in that you don't have to list and name an attribute but rather rely on an implicit 'value' attribute. Still, I propose to make that attribute explicit. New syntax would be:
+This syntax is smart in that you don't have to list and name an attribute but rather rely on an implicit 'value' attribute.
+
+The current syntax, in which an underlying value type is referenced through 'oftype', it does not use instantiation/inheritance but rather composition.
+The purpose of using composition is that it is
+
+1. more similar to traditional approaches and
+2. is needed anyway for multi-attribute value types.
+
+Thus, this RFC removes the `oftype` keyword in a value type definitions and proposes an attribute-based syntax that is extendable to a future multi-attribute syntax.
+
+## Explanation
+
+I propose to make that attribute explicit. The new syntax would be:
 
 ```jayvee
 valuetype CorrelationCoefficient {
-  attributes: [
-    value oftype decimal;
-  ];
-  constraints: [
-    MinusOneToPlusOneRange; // Don't know how to attach this to value attribute
-  ];
+  property correlation oftype decimal;
+
+  constraint minusOneToPlusOneRange: MinusOneToPlusOneRange(value = correlation);
```

Comment:
Not entirely sure if we really agreed on the syntax to call a constraint on a property

Reply:
I don't think we did (and I don't think we have a `constraint` keyword somewhere that is not a constraint definition, right? I'd keep the suggestions in this PR more minimal with 

```
constraints: [
  MinusOneToPlusOneRange on correlation
]
```

I think that nicely aligns with our current syntax style (so doesn't introduce new concepts like constructor-like brackets and assignments). We will of course have to evolve on that (e.g. constraints over multiple attributes, AND/ORing constraints etc) but I'd do that in a later RFC.

Reply:
How about the following?

```
constraint MinusOneToPlusOneRange on correlation;
```

I'd really like to get rid of the array syntax and more closely align with the "flat over nested" design principle.

Reply:
You mean similar to the `property` syntax of just listing all the properties in that way? I am good with it, happy with your suggestion there.

Reply:
Yes exactly. I feel like, currently, we have two different approaches to defining lists. And I'd like to introduce more consistency by also using the same flat syntax as for `property`